### PR TITLE
Fixed multiple particle issues: division by zero, color ramp override, scale dependent on amount ratio

### LINKED ISSUE #83258: Particles emit by GPUParticles2D disappear when tree is paused
### Godot version

v4.2.beta1.official [b1371806a]

### System information

Godot v4.2.beta1 - Windows 10.0.22621 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1660 Ti (NVIDIA; 31.0.15.3699) - AMD Ryzen 7 4800H with Radeon Graphics (16 Threads)

### Issue description

GPUParticles2D that have process mode set to PROCESS_MODE_PAUSABLE disappear when the tree is paused (using `get_tree().paused = true`). The entire particle system restarts when the tree is then unpaused (with `get_tree().paused = false`). as seen in this video.

https://github.com/godotengine/godot/assets/717574/17a66bca-6759-456d-9c59-c589b228bbe6

Expected result is for the particles to stop moving and resume their velocity when the system is unpaused.

### Steps to reproduce

- Open the attached reproduction project
- Open test_scene.tscn and run the scene.
- Observe how both particle systems are playing properly
- Press space bar to pause the game and observe that the particles of the left particle system disappear.
- Press space bar again to unpause, and observe how the left particle system restarts emission.

### Minimal reproduction project

[GPUParticlesTest.zip](https://github.com/godotengine/godot/files/12892432/GPUParticlesTest.zip)

### PR DESCRIPTION
**Pull request by QbieShay**
Closes  https://github.com/godotengine/godot/issues/83258
Fixes scale and amount ratio being tied together
Fixes initial ramp overriding color

The last two issues were reported to me in DM

### CODE REVIEW

File: scene/resources/particle_process_material.cpp [hunk 1]
```diff
@@ -773,18 +773,18 @@ void ParticleProcessMaterial::_update_shader() {
 	code += "		radial_displacement = normalize(radial_displacement) * min(abs((radial_displacement_multiplier * param.radial_velocity)), length(transform[3].xyz - global_pivot));\n";
 	code += "		}\n";
 	code += "	\n";
-	code += "	return radial_displacement * delta;\n";
+	code += "	return radial_displacement;\n";
 	code += "}\n";
 	if (tex_parameters[PARAM_DIRECTIONAL_VELOCITY].is_valid()) {
-		code += "vec3 process_directional_displacement(DynamicsParameters param, float lifetime_percent,mat4 transform, mat4 emission_transform, float delta){\n";
+		code += "vec3 process_directional_displacement(DynamicsParameters param, float lifetime_percent,mat4 transform, mat4 emission_transform,){\n";
```
Comment (AThousandShips) [inline scene/resources/particle_process_material.cpp:779]:
```suggestion
		code += "vec3 process_directional_displacement(DynamicsParameters param, float lifetime_percent,mat4 transform, mat4 emission_transform) {\n";
```

**Comment by Ansraer** [comment]
Discussed during the meeting today, we have concerns that the current random function generates numbers in a repeatable pattern. This is something we would ideally like to investigate before merging.

**Review by Ansraer — APPROVED** [review]

**Review by clayjohn — APPROVED** [review]
Discussed in this week's rendering meeting. Everything looks good. The concern I had over moving the ``AMOUNT_RATIO`` check turned out to be unfounded. Moving ``AMOUNT_RATIO`` is the correct fix

**Review by AThousandShips — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**